worker/pkg/zip/zip.go

```diff
@@ -0,0 +1,48 @@
+package zip
+
+import (
+	"archive/zip"
+	"io"
+)
+
+type Unzipper struct {
+	r   *zip.Reader
+	wFn func(name string) (io.Writer, error)
+}
+
+func NewUnzipper(ra io.ReaderAt, size int64, wFn func(name string) (io.Writer, error)) (*Unzipper, error) {
+	r, err := zip.NewReader(ra, size)
+	if err != nil {
+		return nil, err
+	}
+	return &Unzipper{
+		r,
+		wFn,
+	}, nil
+}
+
+func (uz *Unzipper) Unzip() error {
+	for _, f := range uz.r.File {
+
+		if f.FileInfo().IsDir() {
+			continue
+		} else {
+			rc, err := f.Open()
+			if err != nil {
+				return err
+			}
+			defer rc.Close()
+
+			w, err := uz.wFn(f.FileInfo().Name())
+			if err != nil {
+				return err
+			}
+			_, err = io.Copy(w, rc)
```

Comment:
Zip bomb protection is needed.

```suggestion
			_, err = io.CopyN(w, rc, limit)
			_ = w.Close()
			if errors.Is(io.EOF, err) {
				return &LimitError{ Path: f.FileInfo().Name() }
			}
```

with 

```go
const limit = 1024 * 1024 * 1024 * 10 // 10GB

type LimitError struct {
  Path string
}

func (e *LimitError) Error() string {
  return fmt.Sprtinf("file size limit reached at %s", e.Path)
}
```